app/controllers/orders_controller.rb

```diff
@@ -326,7 +314,10 @@ def index
   # Group into chunks by the group id. If group_id is nil, then it should get its
   # own chunk.
   def grouped_order_details
-    @order.order_details.order(:group_id).slice_when do |before, after|
+    # Do not do any additional scopes on `@order.order_details` to ensure the list
+    # of OrderDetail objects is memoized. Validations are run on `@order.order_details`
+    # and we want to be able to render validation errors inline.
+    @order.order_details.sort_by(&:group_id).slice_when do |before, after|
       before.group_id.nil? || before.group_id != after.group_id
```

Comment:
The issue has more to do with the API of the validators. They take an Order and drop errors on the association's entries. We could change that API to take an array of orderdetails, but that means we'd need to make additional changes downstream in NU. Also, the validator in NU (for research safety) needs access to the order (for `ordered_on_behalf_of?`.

Not quite sure how to make it better.